### PR TITLE
ApiReference docs improvements

### PR DESCRIPTION
**Pull request by Sarabadu**
## Hey, I just made a Pull Request!

This PR aims to add some improvements to the apiReference documentation:

### db00b1a91de7fdb0b9961d263f56a6c761821842 writes the breadcrumbs with backticks:
  - show all but `Home` names in breadcrumbs monospaced
  - avoid spellchecking on breadcrumbs

|Previous | 
|---|
| image|
|image|

| New |
 |---|
 |  image |
|image|

### f17709fa7710c68932d188e426294730347fed3b add a new custom doc annotation to show related configs needed 

| Api reference Docs |
|---|
|image|

| code comments |
|---|
|image|

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/repo-tools/src/commands/api-reports/api-extractor.ts [hunk 1]
```diff
@@ -861,6 +886,32 @@ export async function buildDocs({
...
+            // TODO @sarabadu understand if we need `codeDestination` at all on this custom DocCodeSpanLink
+            super.writeLinkTagWithCodeDestination(node, context);
+          } else if (node.urlDestination) {
+            const linkText =
+              node.linkText !== undefined ? node.linkText : node.urlDestination;
+            const encodedLinkText = this.getEscapedText(
+              linkText.replace(/\s+/g, ' '),
+            );
+            context.writer.write('[');
+            context.writer.write(`${'`'}${encodedLinkText}${'`'}`);
```
Comment (freben) [inline packages/repo-tools/src/commands/api-reports/api-extractor.ts:908]: ``` `${'`'`${'`'`${'`'`${'`'`${'`'`${'`'`${'`'`${'`'`${'`'`${'`'`${'`'`${'`'`${'`'`${'`'`${'`'`${'`' ```

**Review by freben — COMMENTED** [review]
Looks good to me! @Rugvip anything to add?

File: packages/repo-tools/src/commands/api-reports/api-extractor.ts (hunk 1 above)
Comment (Rugvip) [inline packages/repo-tools/src/commands/api-reports/api-extractor.ts:908]:
```suggestion
            context.writer.write(`\`${encodedLinkText}\``);
```

😅

File: .changeset/slow-eggs-grab.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+'@backstage/repo-tools': minor
+'@backstage/plugin-proxy-backend': patch
+---
+
+Api reference documentation improvements
+
+- breadcrumbs links semantics as code spans
+- new `@config` annotation to describe related config keys
```
Comment (Rugvip) [inline .changeset/slow-eggs-grab.md:9]: Gotta split up this changeset, this message really only makes sense for `repo-tools` right?

**Review by Rugvip — CHANGES_REQUESTED** [review]
Nice, thanks! 😁 

Just a minor changeset nit

File: .changeset/slow-eggs-grab.md (hunk 2 above)
Comment (Sarabadu) [reply .changeset/slow-eggs-grab.md:9]: Yes @Rugvip, you are right this only applies to `repo-tools`.
how do I need to split it? creating a different changeset file for `proxy-backend` is enough?

Comment (Sarabadu) [reply .changeset/slow-eggs-grab.md:8]: done,  let me know if looks ok now

File: .changeset/warm-parents-notice.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-proxy-backend': patch
+---
+
+Add documentation to de api reference
```
Comment (Rugvip) [inline .changeset/warm-parents-notice.md:5]:
```suggestion
Documented the `createRouter` method.
```

**Review by Rugvip — APPROVED** [review]
Yep, that 👍

**Pull request merged**